qlty-coverage/src/transformer.rs

```diff
@@ -151,14 +151,6 @@ pub struct StripPrefix {
     prefix: PathBuf,
 }
 
-impl Default for StripPrefix {
-    fn default() -> Self {
-        Self {
-            prefix: current_dir().unwrap_or_else(|_| PathBuf::from(".")),
```

Comment:
Actually instead of removing this why don't we do 
`prefix: Workspace::assert_within_git_directory()?`
Here and we won't have to make any change to publish or transform at the command level and it'll be in one place instead of two?

Reply:
@marschattha the problem with this approach is that we can't use `?` or any form of error reporting. We would need to manually catch the err and best case log it, or panic, the latter we definitely don't want to be doing, and the former would not allow us to exit out on failure, which we definitely _do_ want to be doing. Because of this, we need to trigger the function in a place where that error can be correctly bubbled up and handled.

Reply:
> Also slight bonus would be you can run coverage commands outside of a git repo (although not sure if that would be desirable) by using strip-prefix flags if need be

I think this would be questionably error-prone if allowed, given that any data we collected in this way would be incorrectly stripped unless the user knew about this setting, which is a big discoverability issue. We could warn, but the basis for qlty coverage publish is that it is used within the qlty cloud ecosystem, so I think it's fair to assume there's at least a git repo.

Reply:
As for changing the return: Default trait does not allow this. We would have to switch it out with a separate function. I suppose it's possible to do this, but by the same token, I think it also makes a lot of sense to require a prefix at the usage level-- it will almost never make sense to pass unstripped paths to our coverage uploader, and so I think it's fair to make this requirement explicit at the top of the call stack and "fail early".

It's possible we would want to rename `strip_prefix` to `root_path` to make that more clear.

Reply:
All this said I think I will make the change in Processor to keep this PR simpler and avoid any potential usage regressions, since technically it is a breaking change.

Reply:
> I think this would be questionably error-prone if allowed, given that any data we collected in this way would be incorrectly stripped unless the user knew about this setting, which is a big discoverability issue.

Well if the user is unaware/doesn't use the `strip-prefix` flag, it will default to using the git directory. Only when the user manually specifies a custom `strip-prefix` path it will not lookup the git directory path and use that instead. But yea not sure if there is a good use case for that, so the limit might be better.

> It's possible we would want to rename strip_prefix to root_path to make that more clear.

Agreed with that, that would probably be more clear.